src/Session/Manager.php

```diff
@@ -0,0 +1,383 @@
+
+    /**
+     * @var string
+     */
+    private string $uniqueId = '';
+
+    /**
+     * Manager constructor.
+     *
+     * @param array options = [
+     *     'uniqueId' => null
+     * ]
+     */
+    public function __construct(array $options = [])
+    {
+        $this->setOptions($options);
+    }
+
+    /**
+     * Destroy/end a session
+     */
+    public function destroy(): void
+    {
+        if (true === $this->exists()) {
+            session_destroy();
+
+            $_SESSION = [];
+        }
+    }
+
+    /**
+     * Check whether the session has been started
+     *
+     * @return bool
+     */
+    public function exists(): bool
+    {
+        return (session_status() === self::SESSION_ACTIVE);
+    }
+
+    /**
+     * Gets a session variable from an application context
+     *
+     * @param string     $key
+     * @param mixed|null $defaultValue
+     * @param bool       $remove
+     *
+     * @return mixed|null
+     */
+    public function get(string $key, $defaultValue = null, bool $remove = false)
+    {
+        $value = null;
+
+        if (false === $this->exists()) {
+            // To use $_SESSION variable we need to start session first
+            return $value;
+        }
+
+        $uniqueKey = $this->getUniqueKey($key);
+        $value     = $_SESSION[$uniqueKey] ?? $defaultValue;
+
+        if (true === $remove) {
+            unset($_SESSION[$uniqueKey]);
+        }
+
+        return $value;
+    }
+
+    /**
+     * Returns the stored session adapter
+     *
+     * @return SessionHandlerInterface
+     */
+    public function getAdapter(): SessionHandlerInterface
+    {
+        return $this->adapter;
+    }
+
+    /**
+     * Returns the session id
+     *
+     * @return string
+     */
+    public function getId(): string
+    {
+        return session_id();
+    }
+
+    /**
+     * Returns the name of the session
+     *
+     * @return string
+     */
+    public function getName(): string
+    {
+        if ('' !== $this->name) {
+            $this->name = session_name();
+        }
+
+        return $this->name;
+    }
+
+    /**
+     * Check whether a session variable is set in an application context
+     *
+     * @param string $key
+     *
+     * @return bool
+     */
+    public function has(string $key): bool
+    {
+        if (false === $this->exists()) {
+            // To use $_SESSION variable we need to start session first
+            return false;
+        }
+
+        $uniqueKey = $this->getUniqueKey($key);
+
+        return isset($_SESSION[$uniqueKey]);
+    }
+
+    /**
+     * Get internal options
+     *
+     * @return array
+     */
+    public function getOptions(): array
+    {
+        return $this->options;
+    }
+
+    /**
+     * Regenerates the session id using the adapter.
+     *
+     * @param bool $deleteOldSession
+     *
+     * @return ManagerInterface
+     */
+    public function regenerateId($deleteOldSession = true): ManagerInterface
```

Comment:
```suggestion
    public function regenerateId(bool $deleteOldSession = true): ManagerInterface
```
to be the same as in `ManagerInterface`